evm-test-helpers/src/contracts/oracle.ts

```diff
@@ -0,0 +1,317 @@
+ * ```solidity
+ *  function cancelOracleRequest(
+ *    bytes32 _requestId,
+ *    uint256 _payment,
+ *    bytes4 _callbackFunc,
+ *    uint256 _expiration
+ *  )
+ * ```
+ * method on an Oracle.sol contract.
+ *
+ * @param runRequest The run request to flatten into the correct order to perform the `cancelOracleRequest` function
+ * @param txOpts Additional ethereum tx options
+ */
+export function convertCancelParams(
+  runRequest: RunRequest,
+  txOpts: TxOptions = {},
+): [string, string, string, string, TxOptions] {
+  return [
+    runRequest.requestId,
+    runRequest.payment,
+    runRequest.callbackFunc,
+    runRequest.expiration,
+    txOpts,
+  ]
+}
+
+/**
+ * Abi encode parameters to call the `oracleRequest` method on the [Oracle.sol](../../../evm/contracts/Oracle.sol) contract.
+ * ```solidity
+ *  function oracleRequest(
+ *    address _sender,
+ *    uint256 _payment,
+ *    bytes32 _specId,
+ *    address _callbackAddress,
+ *    bytes4 _callbackFunctionId,
+ *    uint256 _nonce,
+ *    uint256 _dataVersion,
+ *    bytes _data
+ *  )
+ * ```
+ *
+ * @param specId The Job Specification ID
+ * @param callbackAddr The callback contract address for the response
+ * @param callbackFunctionId The callback function id for the response
+ * @param nonce The nonce sent by the requester
+ * @param data The CBOR payload of the request
+ */
+export function encodeOracleRequest(
+  specId: string,
+  callbackAddr: string,
+  callbackFunctionId: string,
+  nonce: number,
+  data: BigNumberish,
+): string {
+  const oracleRequestSighash = '0x40429946'
+  const oracleRequestInputs = [
+    { name: '_sender', type: 'address' },
+    { name: '_payment', type: 'uint256' },
+    { name: '_specId', type: 'bytes32' },
+    { name: '_callbackAddress', type: 'address' },
+    { name: '_callbackFunctionId', type: 'bytes4' },
+    { name: '_nonce', type: 'uint256' },
+    { name: '_dataVersion', type: 'uint256' },
+    { name: '_data', type: 'bytes' },
+  ]
+
+  const encodedParams = ethers.utils.defaultAbiCoder.encode(
+    oracleRequestInputs.map(i => i.type),
+    [
+      ethers.constants.AddressZero,
+      0,
+      specId,
+      callbackAddr,
+      callbackFunctionId,
+      nonce,
+      1,
+      data,
+    ],
+  )
+
+  return `${oracleRequestSighash}${stripHexPrefix(encodedParams)}`
+}
+
+/**
+ * Extract a javascript representation of a run request from the data
+ * contained within a EVM log.
+ * ```solidity
+ *  event OracleRequest(
+ *    bytes32 indexed specId,
+ *    address requester,
+ *    bytes32 requestId,
+ *    uint256 payment,
+ *    address callbackAddr,
+ *    bytes4 callbackFunctionId,
+ *    uint256 cancelExpiration,
+ *    uint256 dataVersion,
+ *    bytes data
+ *  );
+ * ```
+ *
+ * @param log The log to extract the run request from
+ */
+export function decodeRunRequest(log?: ethers.providers.Log): RunRequest {
+  if (!log) {
+    throw Error('No logs found to decode')
+  }
+
+  const types = [
```

Comment:
How do you feel about using uppercase so that it's consistent with other const variables like `SERVICE_AGREEMENT_TYPES`?